src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/validation/AbstractConfigurationValidator.java

```diff
@@ -188,6 +191,42 @@ public boolean validate() {
             return false;
         }
 
+        //null element in the values of all the possible keys with DataType as ARRAY
+        for (Entry<String, DataType> allowedKey : allowedKeys.entrySet()) {
+            DataType dataType = allowedKey.getValue();
+            JsonNode value = contentAsNode.get(allowedKey.getKey());
+            if (dataType == DataType.ARRAY && value != null) {
+                try {
+                    List contentArray = DefaultObjectMapper.objectMapper.convertValue(value, java.util.List.class);
+                    if (contentArray.contains(null)) {
+                        this.errorType = ErrorType.NULL_ARRAY_ELEMENT;
+                        return false;
+                    }
+                    else {
+                        for (int i = 0; i < contentArray.size(); i++) {
+                            if (contentArray.get(i) instanceof HashMap) {
```

Comment:
What if it is more deeply nested structure? Can you work directly with `JsonNode` and `JsonArray`?

Reply:
Done